packages/amplication-data-service-generator/src/util/ast.ts

```diff
@@ -769,3 +779,61 @@ export function createGenericArray(
     builders.tsTypeParameterInstantiation([itemType])
   );
 }
+
+export function deleteDecoratorPropertyByName(
+  node: ASTNode,
+  decoratorName: string
+): namedTypes.Decorator {
+  let decorator: namedTypes.ClassDeclaration | null = null;
+  recast.visit(node, {
+    visitDecorator(path) {
+      const callee = path.get("expression", "callee");
+      if (callee.value && callee.value.property?.name === decoratorName) {
+        decorator = path.value;
+        path.prune();
+      }
+      return this.traverse(path);
+    },
+  });
+
+  if (!decorator) {
+    throw new Error(
+      `Could not find class decorator with the name ${decoratorName} in provided AST node`
+    );
+  }
+
+  return decorator;
+}
+
+export function setDecoratorState(
```

Comment:
We better use a more specific version of this function instead of this generic version.
I think we need to create a function named "removeIdentifierFromUseInterceptorDecorator" in "nestjs-code-generation.ts" 

Different decorators use different constructors and accept different parameters, so this version will not fit other use cases. 

Also, you are expecting a format of a @common.useInterceptor and any other use case where the decorator will not use a namespace, it will not work - so specific version of the function will resolve this issue